### DISCUSSION TITLE
How insertion sort and selection sorts are different?

### DISCUSSION THREAD

**Discussion by ankit-akash** [ankit-akash/DSA#37] [Q&A] [answered]
Brief

**Comment by AlexMi-Ha** [comment, marked as answer]
There are a few differences in both how it works and the runtime:

Insertion Sort:
Insertion sort takes a element and wants to find the spot in the list part before it, where it fits.
So it runs back through the list until it is not smaller than the element on its left anymore.
[6 | 7, 1, 8, 4, 3] -> 7 is bigger than 6 so move on
[6, 7 | 1, 8, 4, 3] -> move 1 to the left until the left element is smaller
[1, 6, 7 | 8, 4, 3] -> and so on.....
Selection Sort:
It searches through the list and tries to find the smallest number -> This number will be put at the first spot of the list.
Then the algorithm is repeated for the rest of the array. Of course the first spot hasn't to be looked at again, as it is already sorted.
[6, 7, 1, 8, 4, 3] -> 1 is smallest so put it to the front
[1 | 6, 7, 8, 4, 3] -> now 3 is the smallest number of the new array
[1, 3 | 6, 7, 8, 4] -> repeat ...

When it comes to Time complexity Insertion Sort performs better

| |	Best|	Average	|Worst|
|-|-|-|-|
|Selection Sort	|Ω(n^2)	|θ(n^2)	|O(n^2)|
|Insertion Sort	|Ω(n)	|θ(n^2)	|O(n^2)|
